jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -62,6 +72,73 @@
     }
   }
 
+  /**
+   * Retrieves the cached manifest for an image reference.
+   *
+   * @param imageReference the image reference
+   * @return the manifest for the image reference, if found
+   * @throws IOException if an I/O exception occurs
+   */
+  Optional<ManifestTemplate> retrieveManifest(ImageReference imageReference) throws IOException {
+    Path imageDirectory = cacheStorageFiles.getImageDirectory(imageReference);
+    Path manifestPath = imageDirectory.resolve("manifest.json");
+    if (!Files.exists(manifestPath)) {
+      return Optional.empty();
+    }
+
+    try (LockFile ignored = LockFile.lock(imageDirectory.resolve("lock"))) {
+      // TODO: Consolidate with ManifestPuller
+      ObjectNode node =
+          new ObjectMapper().readValue(Files.newInputStream(manifestPath), ObjectNode.class);
+      if (!node.has("schemaVersion")) {
```

Comment:
I think in these cases, it means cache corruption, right? So I think we should quit. I am fine with crashing with a runtime exception. Maybe with a message to report the bug that Jib corrupted the cache and a way to clear the cache?